rust/lance-encoding/src/encoder.rs

```diff
@@ -787,6 +791,20 @@ impl CompressionStrategy for CoreArrayEncodingStrategy {
                 return Ok(Box::new(BitpackMiniBlockEncoder::default()));
             }
         }
+        if let DataBlock::VariableWidth(ref variable_width_data) = data {
+            if variable_width_data.bits_per_offset == 32 {
+                let max_len = data
+                    .get_stat(Stat::MaxLength)
+                    .expect("VariableWidthDataBlock should have valid max length statistics");
+                let max_len = max_len
+                    .as_any()
+                    .downcast_ref::<PrimitiveArray<UInt64Type>>()
+                    .unwrap();
+                if max_len.value(0) < 128 {
```

Comment:
Why verify `max_len` again here?  It's not harmful, just curious.

Reply:
it's just being cautious, and subconsciously the fear of writing bug code. I think it's better to remove it.

Reply:
removed.